### ISSUE TITLE
cannot read property push of undefined

### ISSUE THREAD

**Issue by ralphtheninja** [joehand/dat-git-example#1]
```
lms@x240 ~/tmp/sparse-clone/dat-clone-sparse-test (master)
$ dat sync
/home/lms/src/datproject/dat/src/lib/archive.js:42
      if (err) return state.warnings.push(err.message)
                                    ^

TypeError: Cannot read property 'push' of undefined
    at /home/lms/src/datproject/dat/src/lib/archive.js:42:37
    at /home/lms/src/datproject/dat/node_modules/hyperdrive/index.js:543:55
    at /home/lms/src/datproject/dat/node_modules/append-tree/index.js:110:23
    at /home/lms/src/datproject/dat/node_modules/append-tree/index.js:362:10
    at /home/lms/src/datproject/dat/node_modules/append-tree/index.js:372:5
    at /home/lms/src/datproject/dat/node_modules/hypercore/index.js:119:15
    at apply (/home/lms/src/datproject/dat/node_modules/thunky/index.js:44:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by ralphtheninja** [comment]
prob related https://github.com/datproject/dat/issues/838

**Comment by joehand** [comment]
Fixed in dat 13.8.1
